compiler/plugins/target/AMD-AIE/iree-amd-aie/Transforms/AMDAIELogicalObjFifoSplittingUtils.cpp

```diff
@@ -289,16 +353,13 @@ LogicalResult splitLogicalObjectFifos(
   toBeErased.insert(sourceAllocOp);
   toBeErased.insert(sourceObjectFifo);
 
-  SmallVector<OpFoldResult, 4> staticL2AsTargetOffsets =
+  SmallVector<OpFoldResult> staticL2AsTargetOffsets =
       l3ToL2DmaOp.getTargetMixedOffsets();
-  SmallVector<OpFoldResult, 4> staticL2AsTargetSizes =
+  SmallVector<OpFoldResult> staticL2AsTargetSizes =
       l3ToL2DmaOp.getTargetMixedSizes();
-  SmallVector<int64_t, 4> l2ShapeAsTarget = llvm::to_vector(
-      cast<MemRefType>(l3ToL2DmaOp.getTargetObjectFifo().getMemref().getType())
-          .getShape());
-  SmallVector<OpFoldResult, 4> staticL3AsSourceOffsets =
+  SmallVector<OpFoldResult> staticL3AsSourceOffsets =
       l3ToL2DmaOp.getSourceMixedOffsets();
-  SmallVector<OpFoldResult, 4> staticL3AsSourceSizes =
+  SmallVector<OpFoldResult> staticL3AsSourceSizes =
```

Comment:
Relaying here the review comment : https://github.com/nod-ai/iree-amd-aie/pull/755#discussion_r1750237180 (@jtuyls )

Two reasons :-
1. The second argument to `SmallVector<arg1, arg2>` used to be mandatory but stopped being so - I came across this relaxed constraint in one of the upstream contribution's review comments.
2. In case of "splitting" we start creation of a new L2 logicalobjectFifo via L2->L1's route. In case of "combing" we start it from L3->L2. The function `createNewLogicalObjectFifo` declaration complains because :-
  a. `SmallVector<arg1>` != `SmallVector<arg1, 4>`
  b. `SmallVector<arg1, 4>` != `SmallVector<arg1, 6>`
  
Therefore the only way to circumvent point 2z, that I knew of, was to remove this which aligned well with point 1.

Reply:
I think it's still useful to specify the number of elements in the vector as we know it will usually be less than 4 and not much more: `SmallVector<OpFoldResult, 4>`. `createNewLogicalObjectFifo` then has to accept a `SmallVectorImpl` instead of a `SmallVector` to avoid the issue in 2).

Reply:
Sure, I've addressed it in the latest push. Please take a look.